src/devices/machine/mm5740.cpp

```diff
@@ -100,32 +107,27 @@ TIMER_CALLBACK_MEMBER(mm5740_device::perform_scan)
 
 	for (int x = 0; x < 9; x++)
 	{
-		uint16_t data = m_read_x[x]() ^ 0x3ff;
-
-		if ((data ^ m_x_mask[x]) == 0)
-		{
-			// bail early if nothing has changed.
-			continue;
-		}
+		u16 data = m_read_x[x]() ^ 0x3ff;
 
 		for (int y = 0; y < 10; y++)
```

Comment:
The changes to the logic here are dubious.  You should still skip the scan loop if the data hasn’t changed.  The encoder just generates additional data strobe pulses when it detects positive edges on the repeat clock while the data output is valid.

Also, the repeat input is only sampled on falling edges of the clock, and there’s a two cycle latency from detecting a rising edge on the repeat input to generating a data strobe pulse (see fifth page of the datasheet).

Reply:
I saw that there was a 2 cycle latency, but due to the `calc_effective_clock_key_debounce()` calculation, the chip is only operating at 25 Hz. So it would be a much bigger effort to accurately model the 2 cycle delay.

Reply:
Based on testing with the current the capacitance at 5000 pf (as shown in the schematics for terminal board), which provides the effective clock of 25 Hz, seems to behave differently than a real H19/H89. When playing some games, it was missing fast key presses, something that I've never noticed on a real system.  The datasheet for the MM5740, shows that it is linear between 0 pf -> 500 pf, going from 0 mSec -> 4mSec. I'm wondering if it is still linear all the way to 5000 pf. 25Hz seems _really_ low slow for key handling, and seems questionable that would be what it was designed as.